src/schema.rs

```diff
@@ -501,6 +513,46 @@ mod schema_tests {
                         "#),
                 "container_length_type"
         ),
+        case::byte_length_constraint(
+                load(r#"
+                            {{"12345"}}
+                            {{ aGVsbG8= }}
+                        "#),
+                load(r#"
+                            null
+                            null.bool
+                            null.null
+                            null.clob
+                            null.blob
+                            {{}}
+                            {{"1234"}}
+                            {{"123456"}}
+                        "#),
+                load_schema_from_text(r#" // For a schema with byte_length constraint as below:
+                                type::{ name: byte_length_type, byte_length: 5 }
+                        "#),
+                "byte_length_type"
+        ),
+        case::codepoint_length_constraint(
+                load(r#"
+                            '12345'
+                            "12345"
+                        "#),
+                load(r#"
+                            null
+                            null.bool
+                            null.null
+                            null.string
+                            null.symbol
+                            ""
+                            '1234'
+                            "123456"
+                        "#),
+                load_schema_from_text(r#" // For a schema with codepoint_length constraint as below:
+                                type::{ name: codepoint_length_type, codepoint_length: 5 }
```

Comment:
Please add some non-ascii characters (e.g. emoji, Kanji, Hangul) to this test to demonstrate that it works when the codepoint length is different from the utf8 byte length. I believe it will fail using the current implementation.